File: Frontend/RPSGameFrontend/src/app/cores/enums/Message.ts
import { PlayerMoves } from "./playerMoves";

export interface Message 
{
  firstPlayerMoves: PlayerMoves | null;
  secondPlayerMoves: PlayerMoves | null;
  currentRoundNum: number;
  currentRoundWinnerID: string | null; 
  gameWinnerId: string | null;
}

File: Backend/Microservices/GameService/src/Game.Application/DTOs/Message.cs

using Game.Domain.Enums;

namespace Game.Application.DTOs
{
    public class Message
    {
        public PlayerMoves? FirstPlayerMoves { get; set; }
        public PlayerMoves? SecondPlayerMoves { get; set; }
        public int CurrentRaundNum { get; set; }
        public Guid? CurrentRoundWinnerID { get; set; }
        public Guid? GameWinnerId { get; set; }
    }
}
